File: projetoES3/src/model/domain/Cartao.java
package model.domain;
public class Cartao {

	private String numeroCartao;
	private String nomeCartao;
	private String bandeiraCartao;
	private int codigoSegCartao;

	public Cartao(String numeroCartao, String nomeCartao, String bandeiraCartao, int codigoSegCartao) {
		this.numeroCartao = numeroCartao;
		this.nomeCartao = nomeCartao;
		this.bandeiraCartao = bandeiraCartao;
		this.codigoSegCartao = codigoSegCartao;
	}

	public String getNumeroCartao() {
		return numeroCartao;
	}

	public void setNumeroCartao(String numeroCartao) {
		this.numeroCartao = numeroCartao;
	}

	public String getNomeCartao() {
		return nomeCartao;
	}

	public void setNomeCartao(String nomeCartao) {
		this.nomeCartao = nomeCartao;
	}

	public String getBandeiraCartao() {
		return bandeiraCartao;
	}

	public void setBandeiraCartao(String bandeiraCartao) {
		this.bandeiraCartao = bandeiraCartao;
	}

	public int getCodigoSegCartao() {
		return codigoSegCartao;
	}

	public void setCodigoSegCartao(int codigoSegCartao) {
		this.codigoSegCartao = codigoSegCartao;
	}

	
}

File: typeScript/src/model/domain/Cartao.ts
import {EntidadeDominio} from "./EntidadeDominio";
import {Fornecedor} from "./Fornecedor";

export class Cartao extends EntidadeDominio{

    constructor(
        private numeroCartao: string,
        private nomeCartao: string,
        private bandeiraCartao: string,
        private codigoSegCartao: number,
        private fornecedorId: string
    ) {super();}


    public getfornecedorId(): string {
        return this.fornecedorId;
    }

    public setfornecedorId(fornecedorId: string): void {
        this.fornecedorId = fornecedorId;
    }
    public getNumeroCartao(): string {
        return this.numeroCartao;
    }

    public setNumeroCartao(numeroCartao: string): void {
        this.numeroCartao = numeroCartao;
    }

    public getNomeCartao(): string {
        return this.nomeCartao;
    }

    public setNomeCartao(nomeCartao: string): void {
        this.nomeCartao = nomeCartao;
    }

    public getBandeiraCartao() {
        return this.bandeiraCartao;
    }

    public setBandeiraCartao(bandeiraCartao: string): void {
        this.bandeiraCartao = bandeiraCartao;
    }

    public getCodigoSegCartao(): number {
        return this.codigoSegCartao;
    }

    public setCodigoSegCartao(codigoSegCartao: number) {
        this.codigoSegCartao = codigoSegCartao;
    }
}